auth0/src/main/java/com/auth0/android/authentication/storage/SecureCredentialsManager.kt

```diff
@@ -318,82 +323,83 @@ public class SecureCredentialsManager @VisibleForTesting(otherwise = VisibleForT
+            val json: String
+            try {
+                json = String(crypto.decrypt(encrypted))
+            } catch (e: IncompatibleDeviceException) {
+                callback.onFailure(
+                    CredentialsManagerException(
+                        String.format(
+                            "This device is not compatible with the %s class.",
+                            SecureCredentialsManager::class.java.simpleName
+                        ), e
+                    )
                 )
-            )
-            decryptCallback = null
-            return
-        } catch (e: CryptoException) {
-            //If keys were invalidated, existing credentials will not be recoverable.
-            clearCredentials()
-            callback.onFailure(
-                CredentialsManagerException(
-                    "A change on the Lock Screen security settings have deemed the encryption keys invalid and have been recreated. " +
-                            "Any previously stored content is now lost. Please, try saving the credentials again.",
-                    e
+                decryptCallback = null
+                return@execute
+            } catch (e: CryptoException) {
+                //If keys were invalidated, existing credentials will not be recoverable.
+                clearCredentials()
+                callback.onFailure(
+                    CredentialsManagerException(
+                        "A change on the Lock Screen security settings have deemed the encryption keys invalid and have been recreated. " +
+                                "Any previously stored content is now lost. Please, try saving the credentials again.",
+                        e
+                    )
                 )
+                decryptCallback = null
+                return@execute
+            }
+            val bridgeCredentials = gson.fromJson(json, OptionalCredentials::class.java)
+            /* OPTIONAL CREDENTIALS
+             * This bridge is required to prevent users from being logged out when
+             * migrating from Credentials with optional Access Token and ID token
+             */
+            val credentials = Credentials(
+                bridgeCredentials.idToken.orEmpty(),
+                bridgeCredentials.accessToken.orEmpty(),
+                bridgeCredentials.type.orEmpty(),
+                bridgeCredentials.refreshToken,
+                bridgeCredentials.expiresAt ?: Date(),
+                bridgeCredentials.scope
+            )
+            val cacheExpiresAt = storage.retrieveLong(KEY_CACHE_EXPIRES_AT)
+            val expiresAt = credentials.expiresAt.time
+            val hasEmptyCredentials =
+                TextUtils.isEmpty(credentials.accessToken) && TextUtils.isEmpty(credentials.idToken) || cacheExpiresAt == null
+            if (hasEmptyCredentials) {
+                callback.onFailure(CredentialsManagerException("No Credentials were previously set."))
```

Comment:
Is "Credentials" here referring to an actual type? If not it should be lowercase. Same for all the other instances.

Reply:
Yep, Credentials refers to this class here - https://github.com/auth0/Auth0.Android/blob/main/auth0/src/main/java/com/auth0/android/result/Credentials.kt